app/plugins/Files/js/buttons.js

```diff
@@ -177,13 +177,24 @@ eID('upload-dir').onclick = function() {
 	uploadDir(loadPath, nickname);
 };
 
-/*
-eID('search-bar').keydown(function(e) {
-	// TODO: figure out search bar
-	updateFileList(lastLoadedFiles);
-};
-eID('search').onclick = function(e) {
-	// TODO: figure out search bar
-	updateFileList(lastLoadedFiles);
-};
-*/
+// Start search when typing in Search field
+eID('search-bar').onkeyup = function() {
+	tooltip('Searching...', this);
+	var searchstr = eID('search-bar').value;
+	filterFileList(searchstr);
+};
+
+// Filter file list by search string
+function filterFileList(searchstr) {
+	NodeList.prototype.forEach = Array.prototype.forEach
```

Comment:
huh, I didn't know that. Missing semicolons has caused me many problems with JS. Go also uses ASI, but it's far more robust because it's done at compile time rather than run time. Probably best to be strict about semicolons in JS.